packages/sdk/src/services/providers.service.ts

```diff
@@ -1,43 +1,53 @@
-import {PublicClient, createPublicClient, http} from 'viem'
+import {EIP1193Provider, PublicClient, createPublicClient, custom, http} from 'viem'
 import {Network} from '@holographxyz/networks'
 
 import {Config} from './config.service'
-import {HolographLogger} from './logger.service'
 import {UnavailableNetworkError} from '../errors/general/unavailable-network.error'
+import {HolographLogger} from './logger.service'
 import {holographToViemChain} from '../utils/transformers'
 
 export class Providers {
   private readonly logger: HolographLogger
+  private readonly _clientProvider?: unknown
   private readonly _providers: Record<number, PublicClient>
   private readonly _networks: Network[]
 
   constructor(private readonly protocolConfig: Config) {
     this.logger = HolographLogger.createLogger({serviceName: Providers.name})
     this._networks = this.protocolConfig.networks
+    this._clientProvider = this.protocolConfig.provider
     this._providers = {}
 
-    this._networks.forEach((network: Network) => {
-      this._providers[network.chain] = createPublicClient({
-        transport: http(network.rpc),
-        chain: holographToViemChain(network.chain),
+    if (this._networks.length > 0) {
+      this._networks.forEach((network: Network) => {
+        this._providers[network.chain] = createPublicClient({
+          transport: http(network.rpc),
```

Comment:
@feliperodriguess this function is `http`, do they have an `https` version?

Reply:
@sogoiii it's https: https://viem.sh/docs/clients/transports/http#http-transport

`http` is just the wrapper name